### PR TITLE
[Inprogress]SB-1165 Dashboard refactoring

### PR DESCRIPTION
**Pull request by nilesh-more**

### CODE REVIEW

File: src/app/private/scripts/controllers/dashboard/orgDashboardController.js [hunk 1]
```diff
@@ -199,17 +111,26 @@ angular.module('playerApp')
             }
           })
             .catch(function () {
-              toasterService.error(apiMessages.ERROR.get)
+              toasterService.error('error')
```
Comment (nilesh-more) [inline src/app/private/scripts/controllers/dashboard/orgDashboardController.js]: @souravdey091 show proper / useful message here

File: src/app/private/scripts/factories/dashboard/queryService.js [hunk 2]
```diff
@@ -0,0 +1,64 @@
+/**
+ * @author Nilesh More <nilesh_m@tekditechnologies.com>
+ */
+
+'use strict'
+
+angular.module('playerApp')
+  .factory('QueryService', ['config', '$q', 'dashboardService',
```
Comment (kochhar) [inline src/app/private/scripts/factories/dashboard/queryService.js]: do not inject dashboards into the query service.

File: src/app/private/scripts/factories/dashboard/rendererService.js [hunk 3]
```diff
@@ -0,0 +1,141 @@
...
+  .factory('rendererService', ['$filter', 'config', '$timeout', 'toasterService',
+    'uuid4',
+    function ($filter, config, $timeout, toasterService, uuid4) {
+      /**
+       * @method Render
+       * @desc - render line chart
+       * @param   {object}  data
+       * @param   {string}  dashboardType
+       */
+      function Render (data, dashboardType) {
```
Comment (kochhar) [inline src/app/private/scripts/factories/dashboard/rendererService.js]: Please split passing of data into the service and rendering of the chart into separate functions.

File: src/app/private/scripts/services/adapters/announcementAdapter.js [hunk 4]
```diff
@@ -34,8 +34,11 @@ angular.module('playerApp').service('announcementAdapter', ['$rootScope', '$http
     }
 
     function handleHttpRequest (url, data, type, errMsg) {
+      var headers = {
```
Comment (kochhar) [inline src/app/private/scripts/services/adapters/announcementAdapter.js]: please delete headers from the adapter.

File: src/app/private/scripts/services/dashboard/datasources/courseConsumptionDataSource.js [hunk 5]
```diff
@@ -0,0 +1,51 @@
+/**
+ * @author Nilesh More <nilesh_m@tekditechnologies.com>
+ */
+
+'use strict'
+
+angular.module('playerApp')
+  .service('courseConsumptionDataSource', ['$q', '$rootScope', 'config', 'httpAdapter', 'toasterService',
+    'dashboardService', function ($q,
```
Comment (kochhar) [inline src/app/private/scripts/services/dashboard/datasources/courseConsumptionDataSource.js]: please remove dependency on dashboard service.

File: src/test/spec/app/private/scripts/services/dashboard/datasources/courseConsumptionDataSource.js [hunk 6]
```diff
@@ -0,0 +1,58 @@
...
+    $q = _$q_
+    deferred = _$q_.defer()
+    toasterService = _toasterService_
+    httpAdapter = _httpAdapter_
+  }))
+
+  describe('Test course consumption data source service', function () {
+    it('should return valid api response', function () {
+      courseConsDataSource.graphBlockData = []
+      spyOn(httpAdapter, 'httpCall').and.returnValue(deferred.promise)
```
Comment (kochhar) [inline src/test/spec/app/private/scripts/services/dashboard/datasources/courseConsumptionDataSource.js]: shouldn't you assert the url and the request format which is being invoked?

**Comment by kochhar** [comment]
@nilesh-more @souravdey091 please split your PRs and submit them independently.

**Comment by nilesh-more** [comment]
for dashboard data source -  https://github.com/project-sunbird/sunbird-portal/pull/185
for query service - https://github.com/project-sunbird/sunbird-portal/pull/186

**Pull request closed without merging**